pkg/shed/index.go

```diff
@@ -303,21 +310,68 @@ func (f Index) Iterate(fn IndexIterFunc, options *IterateOptions) (err error) {
 			return fmt.Errorf("encode key: %w", err)
 		}
 	}
+
 	it := f.db.NewIterator()
 	defer it.Release()
 
+	var ok bool
+
 	// move the cursor to the start key
-	ok := it.Seek(startKey)
-	if !ok {
-		// stop iterator if seek has failed
-		return it.Error()
+	ok = it.Seek(startKey)
+
+	if !options.Reverse {
+		if !ok {
+			// stop iterator if seek has failed
+			return it.Error()
+		}
+	} else {
+		// reverse seeker
+		if options.StartFrom != nil {
+			if !ok {
+				return it.Error()
+			}
+		} else {
+			// find last key for this index (and prefix)
+
+			// move cursor to last key
+			ok = it.Last()
+			if !ok {
+				return it.Error()
+			}
+
+			if lastKeyHasPrefix := bytes.HasPrefix(it.Key(), prefix); !lastKeyHasPrefix {
+				// increment last prefix byte (that is not 0xFF) to try to find last key
```

Comment:
`(that is not 0xff)` should not be in brackets. `// increment the last prefix byte which is not 0xff to try to find the last key`.

also, i don't really understand why incrementing the last byte which is not `ff` should find the last key. this alters the value from which you would like to seek to (since when you provide the key you would like to start with adds a prefix to the schema prefix, so you might be starting from a wrong item or an item that does not exist). it is not really clear why this brute forcing is needed (or why it should yield the correct item to start with).